pkg/aws/ec2/pricing_map.go

```diff
@@ -127,40 +150,71 @@ func (spm *StructuredPricingMap) GeneratePricingMap(ondemandPrices []string, spo
 	return nil
 }
 
-// AddToPricingMap adds a price to the pricing map. The price is weighted based upon the instance type's CPU and RAM.
-func (spm *StructuredPricingMap) AddToPricingMap(price float64, attribute Attributes) error {
-	spm.m.Lock()
-	defer spm.m.Unlock()
-	if spm.Regions[attribute.Region] == nil {
-		spm.Regions[attribute.Region] = &FamilyPricing{}
-		spm.Regions[attribute.Region].Family = make(map[string]*Prices)
+// GenerateStoragePricingMap receives a json with all the prices of the available storage options
+// It iterates over the storage classes and parses the price for each one.
+func (spm *StoragePricingMap) GenerateStoragePricingMap(storagePrices []string) error {
+	for _, product := range storagePrices {
+		var productInfo storageProduct
+		if err := json.Unmarshal([]byte(product), &productInfo); err != nil {
+			return err
+		}
+
+		region := productInfo.Product.Attributes.Region
+		storageClass := productInfo.Product.Attributes.VolumeApiName
+		if spm.Regions[region] == nil {
+			spm.Regions[region] = &StoragePricing{}
+			spm.Regions[region].Storage = make(map[string]float64)
+		}
+
+		for _, term := range productInfo.Terms.OnDemand {
+			for _, priceDimension := range term.PriceDimensions {
+				price, err := strconv.ParseFloat(priceDimension.PricePerUnit["USD"], 64)
+				if err != nil {
+					log.Printf("error parsing price: %s, skipping", err)
```

Comment:
this is annoying, but my PR review is already going to drive you insane, so might as well throw one more in :)

`StoragePricingMap` should probably define it's own logger so we don't have to use `log.Printf`.  Since this is new code, it makes sense to do it right from the start.  

So something like:

```go 
type StoragePricingMap struct {
	// Regions is a map of region code to StoragePricing
	// key is the region
	// value is a map of storage classes to prices
	Regions map[string]*StoragePricing
	m       sync.RWMutex
	logger *slog.Logger
}

...

func NewStoragePricingMap(l *slog.Logger) *StoragePricingMap {
	return &StoragePricingMap{
		Regions: make(map[string]*StoragePricing),
		m:       sync.RWMutex{},
		logger:  l.With("subsystem", "storagePricing"),
	}
}
```